packages/director/src/lib/hooksReporter.ts

```diff
@@ -0,0 +1,207 @@
+  }
+
+  if (hookEvent === hookEvents.INSTANCE_FINISH) {
+    data.state = 'pending';
+    data.description = `fa:${reportData.currentResults.failures} pa:${reportData.currentResults.passes} pe:${reportData.currentResults.pending} sk:${reportData.currentResults.skipped}`;
+  }
+
+  if (hookEvent === hookEvents.RUN_FINISH) {
+    await new Promise((resolve) => {
+      setTimeout(() => {
+        resolve(null);
+      }, 5000);
+    });
+
+    data.state = 'success';
+    if (reportData.currentResults.failures > 0) {
+      data.state = 'failure';
+    }
+    data.description = `fa:${reportData.currentResults.failures} pa:${reportData.currentResults.passes} pe:${reportData.currentResults.pending} sk:${reportData.currentResults.skipped}`;
+  }
+
+  return (
+    (data.state &&
+      axios({
+        method: 'post',
+        url: fullStatusPostUrl,
+        auth: {
+          username: 'sorry-cypress',
+          password: hook.githubToken,
+        },
+        data,
+      }).catch((err) => {
+        console.error(
+          `Error: Hook post to ${fullStatusPostUrl} responded with `,
+          err
+        );
+      })) ||
+    Promise.resolve()
+  );
+}
+
+export function reportToHook({
+  hookEvent,
+  reportData,
+  project,
+}: {
+  hookEvent: string;
+  reportData: ReportData;
+  project: Project;
+}): Promise<any> {
+  try {
+    reportData.currentResults = getRunTestsOverall(reportData.run);
+    cleanHookReportData(reportData);
+    project &&
+      project.hooks &&
+      project.hooks.forEach((hook) => {
+        if (hook.hookType === hookTypes.GITHUB_STATUS_HOOK) {
+          return reportStatusToGithub({
+            hook,
+            reportData,
+            hookEvent,
+          });
+        } else {
+          if (
+            // if no hooks are specified we should trigger the hook call on all events
+            !hook.hookEvents || //no hooks
+            hook.hookEvents.length < 1 || // no hooks
+            hook.hookEvents.indexOf(hookEvent) !== -1 // matches specific event that fired
+          ) {
+            reportData.hookEvent = hookEvent;
+            reportData.reportUrl = getDashboardRunURL(
+              (reportData.run && reportData.run.runId) ||
+                reportData.instance.runId
+            );
+            return axios({
+              method: 'post',
+              headers: JSON.parse(hook.headers) || {},
+              url: hook.url,
+              data: reportData,
+            }).catch((err) => {
+              console.error(
+                `Error: Hook Post to ${hook.url} responded with `,
+                err
+              );
+            });
+          }
+        }
+      });
+  } catch (error) {
+    console.error(`Failed to run hook at for ${project.projectId}`, error);
+  }
+  return Promise.resolve();
+}
```

Comment:
I hope that works :) Breaking that down would improve readability and review-ability

Reply:
optional chaining would help :)

How are you thinking to break it down? Like should I put the generic webhook code in its own function?

Next to that I commented on what I thought was confusing. What did I miss that I can make less succy?

Reply:
ok, nvm it's more readable at 11am comparing to 1am